gradio/interface.py

```diff
@@ -732,6 +735,9 @@ async def cleanup():
                     batch=self.batch,
                     max_batch_size=self.max_batch_size,
                     concurrency_limit=self.concurrency_limit,
+                    show_progress=cast(
```

Comment:
why is a cast necessary? isn't show_progress set to be of type `Literal["full", "minimal", "hidden"]`

Reply:
pyright infers it as `str` instead of the narrower `Literal` so it complains. I think it's because the property is set in __init__ but it's being used in a separate method here.